### PR TITLE
feat: legal hold indication on user profiles [WPB-4780]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4780" title="WPB-4780" target="_blank">StoryWPB-4780</a>  [Android] Legal hold indication on user profiles
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

As a user I want to see an indication on user profiles that are under legal hold so that I’m more aware of my privacy.

### Solutions

Change the legal hold banner to distinguish between "legal hold active" and "legal hold request pending".
Add legal hold banner to self user profile in both cases and to other user profile when active.
Show proper dialogs when clicking on legal hold banner.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Request legal hold for the current user or enable it for other user and open the self or other user profile to see the banner.

### Attachments (Optional)

https://github.com/wireapp/wire-android/assets/30429749/7dd62048-ff7b-47b2-8bee-64279174af9d

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2861](https://10.10.124.134/job/AR-build-pipeline/2861/) **failed**.

**Comment by AndroidBob** [comment]
Build [2862](https://10.10.124.134/job/AR-build-pipeline/2862/) **failed**.

**Comment by AndroidBob** [comment]
Build [2863](https://10.10.124.134/job/AR-build-pipeline/2863/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8793077)(8793077)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8793077)(8793077)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldBaseBanner.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        modifier = modifier
+            .clip(RoundedCornerShape(dimensions().spacing12x))
+            .background(colorsScheme().surface)
+            .border(
+                width = dimensions().spacing1x,
+                shape = RoundedCornerShape(dimensions().spacing12x),
+                color = colorsScheme().error
+            )
+            .clickable(onClick = onClick)
+            .heightIn(min = 26.dp)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldBaseBanner.kt]: can we have `dimensions()` here as well?

**Review by alexandreferris — APPROVED** [review]
looking great! I just have a tiny comment on a hardcoded dimension

File: app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldBaseBanner.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/legalhold/banner/LegalHoldBaseBanner.kt]: You're right, I missed that 🙈 Done!

**Comment by AndroidBob** [comment]
Build [2870](https://10.10.124.134/job/AR-build-pipeline/2870/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8818111)(8818111)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8818111)(8818111)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]

**Pull request merged**